treasury-fees-monitor/src/agent.js

```diff
@@ -0,0 +1,224 @@
+      const tokenContract = new ethers.Contract(tokenAddress, DECIMALS_ABI, provider);
+      const tokenDecimals = await tokenContract.decimals();
+      data.tokenInfo[tokenAddress] = tokenDecimals.toString();
+    }));
+
+    // get the address of the price oracle
+    const priceOracleAddress = await data.lendingPoolAddressesProviderContract.getPriceOracle();
+    const priceOracleAbi = getAbi(priceOracle.abiFile);
+    data.priceOracleContract = new ethers.Contract(
+      priceOracleAddress,
+      priceOracleAbi,
+      provider,
+    );
+
+    // for each asset retrieve a spot price that will be used when computing statistics
+    const tokenAddressesList = tokenAddresses.map((token) => token[1]);
+    // the spot prices given by the Aave price oracle are in 'ETH wei' units
+    const denominatorWei = (new BigNumber(10)).pow(18);
+    const spotPricesScaledEth = {};
+    // request all of the prices for the assets in tokenAddressesList from the price oracle
+    const spotPricesEth = await data.priceOracleContract.getAssetsPrices(tokenAddressesList);
+    // iterate over each price, correlate it with its associated asset token, and scale the price
+    spotPricesEth.forEach((price, index) => {
+      spotPricesScaledEth[tokenAddressesList[index]] = parseFloat(
+        (new BigNumber(price.toString())).div(denominatorWei),
+      );
+    });
+
+    // parse the csv file specified in the config file and calculate mean, standard deviation,
+    // and variance
+    const {
+      mean, stdDev, variance, numDataPoints,
+    } = await parseCsvAndCompute(config.dataSet, data.tokenInfo, spotPricesScaledEth);
+    data.mean = mean;
+    data.stdDev = stdDev;
+    data.variance = variance;
+    data.numDataPoints = numDataPoints;
+    /* eslint-enable no-param-reassign */
+  };
+}
+
+function provideHandleTransaction(data) {
+  return async function handleTransaction(txEvent) {
+    // check for FlashLoan event, if found then check to see if value is a statistical anomaly given
+    // the current mean + standard deviation (i.e. 3 std deviations or more larger than mean), if so
+    // return a finding, finally re-calculate mean and standard deviation
+    const {
+      protocolName,
+      protocolAbbreviation,
+      developerAbbreviation,
+      lendingPoolAddress,
+      flashLoanSignature,
+      tokenInfo,
+      priceOracleContract,
+      findingLevelThresholds,
+    } = data;
+
+    let {
+      mean,
+      stdDev,
+      variance,
+      numDataPoints,
+    } = data;
+
+    const parsedLogs = txEvent.filterLog(flashLoanSignature, lendingPoolAddress);
+    const findings = (await Promise.all(parsedLogs.map(async (log) => {
+      const finding = [];
+      // scale the found premium using the decimals value in tokenInfo
+      const denominator = (new BigNumber(10)).pow(tokenInfo[log.args.asset]);
+      const scaledPremium = parseFloat(
+        (new BigNumber(log.args.premium.toString())).div(denominator),
+      );
```

Comment:
I think `parseFloat` is potentially risky to use given the size of the numbers.
```javascript
> parseFloat(1000000000000001)
1000000000000001
> parseFloat(10000000000000001)
10000000000000000
> parseFloat(10000000000000000)
10000000000000000
```

```suggestion
      const scaledPremium = new BigNumber(log.args.premium.toString())).div(denominator);
```